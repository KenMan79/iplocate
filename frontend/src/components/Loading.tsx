import React from "react";
import styled, { keyframes } from "styled-components";

const progress = keyframes`
  from {
    width: 0;
  }
  to {
    width: 100%;
  }
`;

const Progress = styled.span`
  display: block;
  animation: ${progress} 3s linear infinite;
  background-color: var(--text-secondary);
  height: 100%;
`;

const Bar = styled.div`
  width: 100%;
  height: 5px;
  position: relative;
  background: var(--text-lightest);
  z-index: 200;
`;

export const Loading: React.FC = () => (
  <Bar>
    <Progress />
  </Bar>
);
