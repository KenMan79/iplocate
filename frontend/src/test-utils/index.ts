export * from "./test-utils";
export * from "./ipdata";
