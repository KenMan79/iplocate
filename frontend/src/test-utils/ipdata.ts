import { IP } from "types";

export const baseIP: IP = {
  continent: {
    code: "EU",
    geonameId: 6255148,
    names: {
      en: "Europe",
    },
  },
  country: {
    geonameId: 3017382,
    isInEuropeanUnion: true,
    isoCode: "FR",
    names: {
      en: "France",
    },
  },
  traits: {
    isAnonymous: false,
    isAnonymousProxy: false,
    isAnonymousVpn: false,
    isHostingProvider: false,
    isLegitimateProxy: false,
    isPublicProxy: false,
    isResidentialProxy: false,
    isSatelliteProvider: false,
    isTorExitNode: false,
    ipAddress: "2.2.2.2",
    network: "2.2.0.0/18",
  },
  city: {},
  location: {
    accuracyRadius: 500,
    latitude: 48.8582,
    longitude: 2.3387,
    timeZone: "Europe/Paris",
  },
};
