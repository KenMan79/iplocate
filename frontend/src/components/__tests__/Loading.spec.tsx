import React from "react";
import { render } from "test-utils";
import { Loading } from "../Loading";

it("matches snapshot", () => {
  const { asFragment } = render(<Loading />);
  expect(asFragment()).toMatchSnapshot();
});
