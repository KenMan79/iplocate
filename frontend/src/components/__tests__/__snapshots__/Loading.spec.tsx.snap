// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: block;
  -webkit-animation: kTNbHI 3s linear infinite;
  animation: kTNbHI 3s linear infinite;
  background-color: var(--text-secondary);
  height: 100%;
}

.c0 {
  width: 100%;
  height: 5px;
  position: relative;
  background: var(--text-lightest);
  z-index: 200;
}

<div
    class="c0"
  >
    <span
      class="c1"
    />
  </div>
</DocumentFragment>
`;
